### ISSUE TITLE
tf.keras.layers.Add with constant fails in tf.keras.models.load_model

### ISSUE THREAD

**Issue by elad-c** [tensorflow/tensorflow#63000]
### Issue type

Bug

### Have you reproduced the bug with TensorFlow Nightly?

Yes

### Source

source

### TensorFlow version

2.15.0

### Custom code

Yes

### Current behavior?

Currently, it fails in building the Add layer:

> [/usr/local/lib/python3.10/dist-packages/keras/src/layers/merging/base_merge.py](https://localhost:8080/#) in <setcomp>(.0)
>      94                 f"Full input_shape received: {input_shape}"
>      95             )
> ---> 96         batch_sizes = {s[0] for s in input_shape if s} - {None}
>      97         if len(batch_sizes) > 1:
>      98             raise ValueError(
> 
> TypeError: unhashable type: 'list'

Expect not to fails...

### Standalone code to reproduce the issue

```shell
import os
import numpy as np
import tempfile
import tensorflow as tf
print(tf.__version__)

x = np.random.randn(1, 2, 2, 3)
_input = tf.keras.layers.Input(x.shape[1:])
const = tf.convert_to_tensor(np.ones((1, 1, 1, 3)).astype(np.float32))
_out = tf.keras.layers.Add()([_input, const])
model = tf.keras.Model(inputs=_input, outputs=_out)

_, tmp_h5_file = tempfile.mkstemp('.keras')
tf.keras.models.save_model(model, tmp_h5_file)
loaded_model = tf.keras.models.load_model(tmp_h5_file)  # <== this line fails
os.remove(tmp_h5_file)
```

### Relevant log output

```shell
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
<ipython-input-3-b3b2b0b93053> in <cell line: 9>()
      7 _, tmp_h5_file = tempfile.mkstemp('.keras')
      8 tf.keras.models.save_model(model, tmp_h5_file)
----> 9 loaded_model = tf.keras.models.load_model(tmp_h5_file)
     10 os.remove(tmp_h5_file)

8 frames
/usr/local/lib/python3.10/dist-packages/keras/src/saving/saving_api.py in load_model(filepath, custom_objects, compile, safe_mode, **kwargs)
    252                 f"with the native Keras format: {list(kwargs.keys())}"
    253             )
--> 254         return saving_lib.load_model(
    255             filepath,
    256             custom_objects=custom_objects,

/usr/local/lib/python3.10/dist-packages/keras/src/saving/saving_lib.py in load_model(filepath, custom_objects, compile, safe_mode)
    279 
    280     except Exception as e:
--> 281         raise e
    282     else:
    283         return model

/usr/local/lib/python3.10/dist-packages/keras/src/saving/saving_lib.py in load_model(filepath, custom_objects, compile, safe_mode)
    244             # Construct the model from the configuration file in the archive.
    245             with ObjectSharingScope():
--> 246                 model = deserialize_keras_object(
    247                     config_dict, custom_objects, safe_mode=safe_mode
    248                 )

/usr/local/lib/python3.10/dist-packages/keras/src/saving/serialization_lib.py in deserialize_keras_object(config, custom_objects, safe_mode, **kwargs)
    726     safe_mode_scope = SafeModeScope(safe_mode)
    727     with custom_obj_scope, safe_mode_scope:
--> 728         instance = cls.from_config(inner_config)
    729         build_config = config.get("build_config", None)
    730         if build_config:

/usr/local/lib/python3.10/dist-packages/keras/src/engine/training.py in from_config(cls, config, custom_objects)
   3328                 # Revive Functional model
   3329                 # (but not Functional subclasses with a custom __init__)
-> 3330                 inputs, outputs, layers = functional.reconstruct_from_config(
   3331                     config, custom_objects
   3332                 )

/usr/local/lib/python3.10/dist-packages/keras/src/engine/functional.py in reconstruct_from_config(config, custom_objects, created_layers)
   1503                 while layer_nodes:
   1504                     node_data = layer_nodes[0]
-> 1505                     if process_node(layer, node_data):
   1506                         layer_nodes.pop(0)
   1507                     else:

/usr/local/lib/python3.10/dist-packages/keras/src/engine/functional.py in process_node(layer, node_data)
   1443                     input_tensors
   1444                 )
-> 1445             output_tensors = layer(input_tensors, **kwargs)
   1446 
   1447             # Update node index map.

/usr/local/lib/python3.10/dist-packages/keras/src/utils/traceback_utils.py in error_handler(*args, **kwargs)
     68             # To get the full stack trace, call:
     69             # `tf.debugging.disable_traceback_filtering()`
---> 70             raise e.with_traceback(filtered_tb) from None
     71         finally:
     72             del filtered_tb

/usr/local/lib/python3.10/dist-packages/keras/src/layers/merging/base_merge.py in <setcomp>(.0)
     94                 f"Full input_shape received: {input_shape}"
     95             )
---> 96         batch_sizes = {s[0] for s in input_shape if s} - {None}
     97         if len(batch_sizes) > 1:
     98             raise ValueError(

TypeError: unhashable type: 'list'
```

**Comment by Venkat6871** [comment]
Hi @elad-c ,
I tried to run your code on Colab using TF v2.15 and nightly versions, In nightly version it is working fine. Could you please check with nightly version. Please find the [gist](https://colab.sandbox.google.com/gist/Venkat6871/503fcbc95d8a64d692d252ac4ca83493/63000_2-15-nightly.ipynb) here for reference.

Thank you!

**Comment by elad-c** [comment]
Thanks!
So I should expect it to be solved in TF 2.17?

**Comment by Venkat6871** [comment]
Hi **@elad-c** ,

Could you please confirm if this issue is resolved for you ? Please feel free to close the issue if it is resolved ? 

Thank you!
